src/tests/ftest/util/agent_utils.py

```diff
@@ -151,11 +148,21 @@ def run_agent(basepath, server_list, client_list=None):
             expected_data += output
 
             match = re.findall(pattern, output)
-            if len(match) > 0:
+            if match:
+                started_clients.append(client)
                 print("<AGENT> agent started on node {} in {} "
                       "seconds".format(client, time.time() - start_time))
                 break
 
+        return_code = sessions[client].poll()
```

Comment:
Given that the previous `.poll()` call in the `while` loop would have already set the return code, would it be slightly more efficient to use this instead:
```suggestion
        return_code = sessions[client].returncode
```

Reply:
Done